core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -74,8 +74,11 @@ class StorageStatusListener(conf: SparkConf) extends SparkListener {
     synchronized {
       val blockManagerId = blockManagerAdded.blockManagerId
       val executorId = blockManagerId.executorId
-      val maxMem = blockManagerAdded.maxMem
-      val storageStatus = new StorageStatus(blockManagerId, maxMem)
+      // This two fields are compatible with old event logs, in which there only has max on heap
+      // memory in the event log. So maxOnHeapMem will use maxMem, maxOffHeapMem will set to 0.
+      val maxOnHeapMem = blockManagerAdded.maxOnHeapMem.getOrElse(blockManagerAdded.maxMem)
```

Comment:
actually wait ... are you sure about this?  it looks to me like in old event logs, we actually dont' know the breakdown between onheap and offheap.  I'm pretty sure the value in old event logs starts from here:

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L154

so then we have to decide how to represent that old info.  The options are:

a) don't show anything, with an empty string in the display, and the types are `Option` even in the rest api
b) show a value of `0` for both of them if they are missing
c) assume all the memory was onheap

this is implementing (c).  I'd prefer (a), if my understanding is correct.